model/src/main/java/org/eclipse/rdf4j/model/impl/LinkedHashModel.java

```diff
@@ -79,8 +79,12 @@ public LinkedHashModel(Collection<? extends Statement> c) {
 
 	public LinkedHashModel(int size) {
 		super();
-		values = new HashMap<>(size * 2);
-		statements = new LinkedHashSet<>(size);
+
+		// assume fewer unique values than statements (eg. some reuse of nodes instatements)
+		values = new HashMap<>(size, 0.75f);
+
+		// compensate for the loadfactor by multiplying by 2
+		statements = new LinkedHashSet<>(size * 2, 0.75f);
```

Comment:
this helps performance, though I don't have any explicit benchmarks just for this